### PR TITLE
add cartesian indexing

### PR DESCRIPTION
**Pull request by tjdiamandis**
Allows for cartesian indexing for variables. Can change this to `AbstractExpr` if preferred.

### CODE REVIEW

File: src/atoms/affine/index.jl [hunk 1]
```diff
@@ -133,4 +133,7 @@ getindex(x::AbstractExpr, cln_r::Colon, col) = getindex(x, 1:size(x)[1], col)
 # All columns for this row(s)
 getindex(x::AbstractExpr, row, cln_c::Colon) = getindex(x, row, 1:size(x)[2])
 
+# Cartesian Index
+Convex.getindex(x::Variable, c::CartesianIndex{2}) = x[c[1], c[2]]
```
Comment (odow) [inline src/atoms/affine/index.jl:137]:
```suggestion
getindex(x::Variable, c::CartesianIndex{2}) = x[c[1], c[2]]
```

**Comment by odow** [comment]
> Can change this to AbstractExpr if preferred.

That would probably be good. Also, what about other `CartesianIndex{N}`?

This also needs tests.

**Comment by tjdiamandis** [comment]
Made the change, but `Convex.jl` seems to only support two dimensional expressions.

https://github.com/jump-dev/Convex.jl/blob/d5e02084dd7a5f7764bf7c738ca9ee8566e26c04/src/expressions.jl#L100

File: src/atoms/affine/index.jl (hunk 1 above)
Comment (tjdiamandis) [reply src/atoms/affine/index.jl]: done

**Comment by odow** [comment]
> seems to only support two dimensional expressions

Also vectors? I haven't looked at the details.

**Comment by tjdiamandis** [comment]
I think as n x 1 arrays 

https://github.com/jump-dev/Convex.jl/blob/d5e02084dd7a5f7764bf7c738ca9ee8566e26c04/src/variable.jl#L187

File: test/test_utilities.jl [hunk 2]
```diff
@@ -409,6 +409,17 @@ end
         @test size(y) == (2, 1)
     end
 
+    @testset "Cartesian index" begin
+        x = Variable(3, 2)
+        for ind in CartesianIndices(zeros(3, 2))
+            @test x[ind] === x[ind[1], ind[2]]
+        end
+        y = [1.0 2 3; 4 5 6] * x
```
Comment (odow) [inline test/test_utilities.jl:417]:
```suggestion
        y = [1.0 2 3; 4 5 6] * x
```

File: test/test_utilities.jl [hunk 3]
```diff
@@ -409,6 +409,17 @@ end
         @test size(y) == (2, 1)
     end
 
+    @testset "Cartesian index" begin
+        x = Variable(3, 2)
+        for ind in CartesianIndices(zeros(3, 2))
+            @test x[ind] === x[ind[1], ind[2]]
+        end
+        y = [1.0 2 3; 4 5 6] * x 
+        for ind in CartesianIndices(zeros(2,2))
```
Comment (odow) [inline test/test_utilities.jl:418]:
```suggestion
        for ind in CartesianIndices(zeros(2, 2))
```

**Review by odow — APPROVED** [review]

**Pull request merged**